app/api/views.py

```diff
@@ -0,0 +1,224 @@
+            )
+
+
+class ValidateResponseMixin:
+    def validate_response_exists(self, url_params: dict[str, str]):
+        """
+        Validates the Response exists for the FeedbackForm
+        """
+
+        response_exists = Response.objects.filter(
+            uuid=url_params["response_id"],
+            feedback_form__uuid=url_params["feedback_form_id"],
+        ).exists()
+
+        if not response_exists:
+            raise NotFound(
+                f"Response id={url_params["response_id"]} does not exist in feedback form id={url_params["feedback_form_id"]}."
+            )
+
+
+class FeedbackFormDetail(generics.RetrieveAPIView):
+    queryset = FeedbackForm.objects.all()
+    serializer_class = FeedbackFormSerializer
+
+    def get_queryset(self):
+        return self.queryset.filter(
+            uuid=self.kwargs["feedback_form_id"],
+            project__uuid=self.kwargs["project_id"],
+        ).prefetch_related(
+            Prefetch(
+                "prompts",
+                queryset=Prompt.objects.select_subclasses(),
+            ),
+        )
+
+    def get_object(self):
+        return generics.get_object_or_404(self.get_queryset())
+
+
+class PromptList(generics.ListAPIView, ValidateFeedbackFormMixin):
+    queryset = Prompt.objects.all()
+    serializer_class = PromptSerializer
+
+    def get_queryset(self):
+        return (
+            self.queryset.filter(
+                feedback_form__uuid=self.kwargs["feedback_form_id"],
+            )
+            .order_by("order")
+            .select_subclasses()
+        )
+
+    def get(self, request, *args, **kwargs):
+        self.validate_feedback_form_exists(kwargs)
+        return super().get(request, *args, **kwargs)
+
+
+class ResponseListCreate(generics.ListCreateAPIView, ValidateFeedbackFormMixin):
+    queryset = Response.objects.all()
+    serializer_class = ResponseSerializer
+
+    def get_queryset(self):
+        return (
+            self.queryset.filter(
+                feedback_form__uuid=self.kwargs["feedback_form_id"]
+            )
+            .select_related("feedback_form")
+            .prefetch_related(
+                Prefetch(
+                    "prompt_responses",
+                    queryset=PromptResponse.objects.select_subclasses().prefetch_related(
+                        "rangedpromptresponse__value"
+                    ),
+                ),
+                Prefetch(
+                    "prompt_responses__prompt",
+                    queryset=Prompt.objects.select_subclasses(),
+                ),
+            )
+        )
+
+    def create(self, request, *args, **kwargs):
+        self.validate_feedback_form_exists(kwargs, ensure_enabled=True)
+
+        request.data["feedback_form"] = kwargs["feedback_form_id"]
+        return super().create(request, *args, **kwargs)
+
+    def list(self, request, *args, **kwargs):
+        self.validate_feedback_form_exists(kwargs)
+        return super().list(request, *args, **kwargs)
```

Comment:
Is there a point adding list endpoints if they won't be used?

Reply:
I think it's quite likely this will be used, TNA have expressed a desire to extract data through the API, and getting responses by feedback form seems like a common use case.

 I also think it will be helpful for them to see the data while integrating.